### PR TITLE
Turn 'sf::Transform' into a 'constexpr' class

### PR DESCRIPTION
**Pull request by vittorioromeo**
Thanks a lot for making a contribution to SFML! 🙂

Before you create the pull request, we ask you to check the follow boxes. (For small changes not everything needs to ticked, but the more the better!)

* [x] Has this change been discussed on [the forum](https://en.sfml-dev.org/forums/index.php#c3) or in an issue before?
* [x] Does the code follow the SFML [Code Style Guide](https://www.sfml-dev.org/style.php)?
* [ ] Have you provided some example/test code for your changes?
* [ ] If you have additional steps which need to be performed list them as tasks!

----

## Description

This PR turns `sf::Transform` into a `constexpr`-friendly class.

## Tasks

* [ ] Tested on Linux
* [x] Tested on Windows
* [ ] Tested on macOS
* [ ] Tested on iOS
* [ ] Tested on Android

## How to test this PR?

CI.

### CODE REVIEW

File: include/SFML/Graphics/Transform.hpp [hunk 1]
```diff
@@ -31,16 +31,17 @@
 #include <SFML/Graphics/Export.hpp>
 #include <SFML/Graphics/Rect.hpp>
 #include <SFML/System/Vector2.hpp>
-#include <SFML/System/Angle.hpp>
 
 
 namespace sf
 {
+class Angle;
+
```
Comment (ChrisThrasher) [inline include/SFML/Graphics/Transform.hpp:39]: Is this trying to minimize compile time? Is that compile time improvement measurable or meaningful?

**Review by ChrisThrasher — COMMENTED** [review]
Can we add a `static_assert` to the unit tests that ensure this can be used at compile time?

**Comment by eXpl0it3r** [comment]
SFMLBuildMaster: Build this please.

**Comment by Bromeon** [comment]
Before we turn SFML into a header-only library, do we know that we'll never want to go back? 🙂 

Removing `constexpr` will be a breaking change. We give up the freedom to change implementation (let's say by using a vector library); at least not without sacrificing a lot of compile time. Maybe that's an OK trade-off, but we should keep it in mind.

Also, while a single `constexpr` class likely makes no meaningful difference in compile time, do we know that this is still the case when a significant part of SFML has its implementation in the header? I assume this is negligible compared to monomorphization (template instantiation) of standard headers?

**Review by Bromeon — APPROVED** [review]
I haven't checked every single line moved into the header, so I assume semantics are unchanged 😀

If we have the buy-in from everyone regarding above-mentioned caveats, this is good to go!

**Pull request merged**